node/pkg/publicrpc/publicrpcserver.go

```diff
@@ -75,6 +76,17 @@ func (s *PublicrpcServer) GetSignedVAA(ctx context.Context, req *publicrpcv1.Get
 	addr := vaa.Address{}
 	copy(addr[:], address)
 
+	// Chain input validation
+	chainId, err := vaa.ChainIDFromInt(uint16(req.MessageId.EmitterChain))
+	if err != nil || chainId == vaa.ChainIDUnset {
+		return nil, status.Error(codes.InvalidArgument, "invalid chain specified")
+	}
+
+	// Sequence input validation
+	if req.MessageId.Sequence == 0 {
+		return nil, status.Error(codes.InvalidArgument, "no sequence specified")
+	}
```

Comment:
```suggestion
```

This logic is invalid as sequence 0 is (much to my chagrin) actually valid.

A quick example is from Aurora's token bridge:
https://github.com/certusone/wormhole/blob/0018bf7250954a56bb8f9953c70f0a44d05be3bf/node/pkg/common/mainnet_consts.go#L74

See this link for the example vaa with sequence 0 on aurora's token bridge: [/v1/signed_vaa/9/00000000000000000000000051b5123a7b0F9b2bA265f9c4C8de7D78D52f510F/0](https://wormhole-v2-mainnet-api.certus.one/v1/signed_vaa/9/00000000000000000000000051b5123a7b0F9b2bA265f9c4C8de7D78D52f510F/0)

And [here is](https://europe-west3-wormhole-315720.cloudfunctions.net/mainnet-readrow?emitterChain=9&emitterAddress=00000000000000000000000051b5123a7b0f9b2ba265f9c4c8de7d78d52f510f&sequence=0000000000000000) the decoded payload which for this vaa.

It's much easier to read with curl and jq:

```
$ curl -sq 'https://europe-west3-wormhole-315720.cloudfunctions.net/mainnet-readrow?emitterChain=9&emitterAddress=00000000000000000000000051b5123a7b0f9b2ba265f9c4c8de7d78d52f510f&sequence=0000000000000000' | jq '.EmitterChain, .Sequence, .TransferDetails'
"aurora"
"0"
{/v1/signed_vaa/9/00000000000000000000000051b5123a7b0F9b2bA265f9c4C8de7D78D52f510F/0  "Amount": ".749002",
  "Decimals": "6",
  "NotionalUSDStr": "0.750372",
  "TokenPriceUSDStr": "1.001829",
  "TransferTimestamp": "2022-04-13 21:18:22 +0000 UTC",
  "OriginSymbol": "UST",
  "OriginName": "UST",
  "OriginTokenAddress": "ust"
}
```

Reply:
Yeah, will probably need to remove this validation, as a nil/omission will default to 0.